arbitrum-docs/stylus/stylus-gas.md

```diff
@@ -0,0 +1,221 @@
+- Memory, which is **100-500x** cheaper due to Stylus's novel exponential pricing mechanism intended to address Vitalik's concerns with the EVM's per-call, [quadratic memory pricing policy](https://notes.ethereum.org/@vbuterin/proposals_to_adjust_memory_gas_costs). For the first time ever, high-memory applications are possible on an EVM-equivalent chain.
+- Storage, for which the Rust SDK promotes better access patterns and type choices. Note that while the underlying `[SLOAD](https://www.evm.codes/#54)` and `[SSTORE](https://www.evm.codes/#55)` operations cost as they do in the EVM, the Rust SDK implements an optimal caching policy that minimizes their use. Exact savings depends on the program.
+- VM affordances, including common operations like `keccak` and reentrancy detection. No longer is it expensive to make safety the default.
+
+There are, however, minor overheads to using Stylus that may matter to your application:
+
+- The first time a WASM is deployed, it must be *activated*. This is generally a few million gas, though to avoid testnet DoS, we've set it to a fixed 14 million. Note that you do not have to activate future copies of the same program. For example, the same NFT template can be deployed many times without paying this cost more than once. We will soon make the fees paid depend on the program, so that the gas used is based on the complexity of the WASM instead of this very conservative, worst-case estimate.
+- Calling a Stylus program costs 128-2048 gas. We're working with Wasmer to improve setup costs, but there will likely always be some amount of gas one pays to jump into WASM execution. This means that if a contract does next to nothing, it may be cheaper in Solidity. However if a contract starts doing interesting work, the dynamic fees will quickly make up for this fixed-cost overhead.
+
+Though conservative bounds have been chosen for testnet, all of this is subject to change as pricing models mature and further optimizations are made. Since gas numbers will vary across updates, it may make more sense to clock the time it takes to perform an operation rather than going solely by the numbers reported in receipts.
+
+# Ink and gas
+
+Because WASM opcodes are orders of magnitude faster than their EVM counterparts, almost every operation that Stylus does costs less than `1 gas`. “Fractional gas” isn’t an EVM concept, so the Stylus VM introduces a new unit of payment known as ink that’s orders of magnitude smaller.
+
+```jsx
+1 gas = 10,000 ink
+```
+
+## Intuition
+
+To build intuition for why this is the case, consider the `ADD` instruction. 
+
+1. Pop two items of the simulated stack
+
```

Comment:
Oops, looks like this should be in the next section :)

Reply:
✅